api/net/Areas/Subscriber/Controllers/ReportController.cs

```diff
@@ -147,31 +161,46 @@ public IActionResult Add(ReportModel model)
         var username = User.GetUsername() ?? throw new NotAuthorizedException("Username is missing");
         var user = _userService.FindByUsername(username) ?? throw new NotAuthorizedException("User does not exist");
         model.OwnerId = user.Id;
+
+        // If there are no subscribers, add the owner as a subscriber.
+        if (!model.Subscribers.Any())
+        {
+            model.Subscribers = new[] { new UserReportModel() { UserId = user.Id, ReportId = model.Id, IsSubscribed = true } };
+        }
+
         var result = _reportService.AddAndSave(model.ToEntity(_serializerOptions));
-        var report = _reportService.FindById(result.Id) ?? throw new InvalidOperationException("Report does not exist");
+        var report = _reportService.FindById(result.Id) ?? throw new NoContentException("Report does not exist");
         return CreatedAtAction(nameof(FindById), new { id = report.Id }, new ReportModel(report, _serializerOptions));
     }
 
     /// <summary>
     /// Update report for the specified 'id'.
     /// </summary>
     /// <param name="model"></param>
+    /// <param name="updateInstances"></param>
     /// <returns></returns>
     [HttpPut("{id}")]
     [Produces(MediaTypeNames.Application.Json)]
     [ProducesResponseType(typeof(ReportModel), (int)HttpStatusCode.OK)]
     [ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.BadRequest)]
     [SwaggerOperation(Tags = new[] { "Report" })]
-    public IActionResult Update(ReportModel model)
+    public IActionResult Update(ReportModel model, [FromQuery] bool updateInstances = false)
     {
         var username = User.GetUsername() ?? throw new NotAuthorizedException("Username is missing");
         var user = _userService.FindByUsername(username) ?? throw new NotAuthorizedException("User does not exist");
-        var result = _reportService.FindById(model.Id) ?? throw new InvalidOperationException("Report does not exist");
+        var result = _reportService.FindById(model.Id) ?? throw new NoContentException("Report does not exist");
         if (result?.OwnerId != user?.Id) throw new NotAuthorizedException("Not authorized to update this report");
         _reportService.ClearChangeTracker(); // Remove the report from context.
-
         result = _reportService.UpdateAndSave(model.ToEntity(_serializerOptions));
-        var report = _reportService.FindById(result.Id) ?? throw new InvalidOperationException("Report does not exist");
+        if (updateInstances && model.Instances.Any())
```

Comment:
I don't like how I implemented this, but I didn't want to make two separate requests to update a report when a subscriber edited their report and instance at the same time.